coalib/bearlib/abstractions/ExternalBearWrap.py

```diff
@@ -0,0 +1,174 @@
+            the decorator in order to create the arguments needed for
+            the executable.
+            """
+            raise NotImplementedError
+
+        @classmethod
+        def get_executable(cls):
+            """
+            Returns the executable of this class.
+
+            :return: The executable name.
+            """
+            return options["executable"]
+
+        @staticmethod
+        def normalize_desc(val):
+            """
+            Normalizes the description of the parameters only if there
+            is none provided.
+
+            :return: A value for the OrderedDict in the FunctionMetada object.
+            """
+            if val[0] != "":
+                return val
+            desc = "No description given."
+            if len(val) == 2:
+                return (desc, val[1])
+            else:
+                desc += " Defaults to " + str(val[2])
+                return (desc, val[1], val[2])
+
+        @classmethod
+        def get_non_optional_params(cls):
+            """
+            Fetches the non_optional_params from ``options['settings']``
+            and also normalizes their descriptions.
+
+            :return: An OrderedDict that is used to create a
+                FunctionMetada object.
+            """
+            non_optional_params = {}
+            for key, value in options['settings'].items():
+                if len(value) == 2:
+                    non_optional_params[key] = cls.normalize_desc(value)
+            return OrderedDict(non_optional_params)
+
+        @classmethod
+        def get_optional_params(cls):
+            """
+            Fetches the optional_params from ``options['settings']``
+            and also normalizes their descriptions.
+
+            :return: An OrderedDict that is used to create a
+                FunctionMetada object.
+            """
+            optional_params = {}
+            for key, value in options['settings'].items():
+                if len(value) == 3:
+                    optional_params[key] = cls.normalize_desc(value)
+            return OrderedDict(optional_params)
+
+        @classmethod
+        def get_metadata(cls):
+            return FunctionMetadata(
+                'run',
+                optional_params=cls.get_optional_params(),
+                non_optional_params=cls.get_non_optional_params())
+
+        @staticmethod
+        def get_severity(result):
+            """
+            Returns the severity corresponding to the one from the
+                parsed JSON
+
+            :param result: A result dict extracted from the parsed JSON
+
+            :return: A RESULT_SEVERITY
+            """
+            if not 'severity' in result or result['severity'] == "NORMAL":
+                return RESULT_SEVERITY.NORMAL
+            elif result['severity'] == "MAJOR":
+                return RESULT_SEVERITY.MAJOR
+            elif result['severity'] == "INFO":
+                return RESULT_SEVERITY.INFO
```

Comment:
ahm it's an Enum, right? It should have some kind of reverse_dict automatically generated that does this translation for you :)